server.py

```diff
@@ -27,34 +29,69 @@ def allowed_file(filename):
     """
     return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
 
+def table_listing_post(request):
+    query, response, status = driver.filter_table(request.json)
+    app.logger.info("Received query: " + str(query))
+    return make_response(jsonify(response), status)
+
+def table_listing_get(request):
+    table_name = request.args.get('table')
+    if table_name is None:
+        return make_response(jsonify('Table name not supplied.'), 400)
+    try:
+        # TODO: once authentication is in place, restrict the tables that can be listed here
+        columns = request.args.get('column')
+        if columns is not None:
+            columns = str.split(columns.strip(), ' ')
+        table_info_obj = driver.get_table(table_name, columns)
+        return make_response(jsonify(table_info_obj), 200)
+    except driver.InvalidTableException:
+        return make_response(jsonify('Table ' + table_name + ' does not exist.'), 404)
+
+def checkTableRouting(table, tableList):
+    if table in tableList:
+        return True, None
+    else:
+        msg = {
+            'message':'Innapropriate route for requested table'
+        }
+        return False, msg
+
+#TODO:Update so it requires authentication?
+@app.route("/adminlist", methods=["GET","POST"])
+def fetch_admin_data(admin=True):
+    """
+    Displays the contents of the table listed in the request. 
+    Usage:
+        GET /list?table=<table_name> to retrieve entire table
+        POST /list to process specific query
+    Returns ({}): JSON object of table data
+    """
+    valid, msg = checkTableRouting(request.args.get('table'), restrictedTables)
+    if not valid:
+        return make_response(jsonify(msg), 404)
+    if request.method == 'POST':
+        return table_listing_post(request)
+    if request.method == 'GET':
+        return table_listing_get(request)
```

Comment:
The 'admin' parameter should be part of the table_listing_post/get functions, not the server routes. Have fetch_admin_data pass 'admin=True' to those functions, while fetch_data doesn't need to pass anything and the table listing functions have a default parameter of 'admin=False'. That way, requests that come via fetch_admin_data can be distinguished from requests that come via fetch_data. Do the check as to whether the table is in the appropriate list there, based on the value of 'admin'.

Reply:
I think this is fixed now. Based off the other comment, checking the table is totally deleted now. Should I reinstate this within the "post"/"get" functions?

Reply:
Yes, we still want to check whether the request came from an admin-validated route. So add back the checkTableRouting functionality, but make a distinction between the tables available to a non-admin request and an admin request. In the table_listing_get/post functions, use the value of 'admin' to verify that the requested table is in the right list. 

You could do this one of two ways: first, you could have the two lists of tables you had in a previous commit, and use nonRestrictedTables as the "acceptable" list for the non-admin path and restrictedTables + nonRestrictedTables as the acceptable list for the admin path.

A better approach though would be to leverage the fact that the driver already has a get_table_list function to create a list of tables that are active in the database. This could serve as the acceptable list for the admin route. You could then have a short list of "admin-only" tables (for now, transactions and archive) and remove these tables from the acceptable list for the non-admin route. This approach requires some extra work and might involve moving the admin-vs-non-admin logic to the driver, but has the benefit that get_table_list doesn't rely on hard-coded lists of tables that can become outdated. In general, we want to avoid this kind of hardcoded list wherever possible, since it increases the need for maintenance and the potential for bugs as the project develops over time.